crates/transaction-pool/src/pool/best.rs

```diff
@@ -10,6 +10,40 @@ use std::{
 };
 use tracing::debug;
 
+/// An iterator that returns transactions that can be executed on the current state (*best*
+/// transactions).
+///
+/// This is a wrapper around [`BestTransactions`] that also enforces a specific basefee.
+///
+/// This iterator guarantees that all transaction it returns satisfy the base fee.
+pub(crate) struct BestTransactionsWithBasefee<T: TransactionOrdering> {
+    pub(crate) best: BestTransactions<T>,
+    pub(crate) base_fee: u128,
+}
+
+impl<T: TransactionOrdering> crate::traits::BestTransactions for BestTransactionsWithBasefee<T> {
+    fn mark_invalid(&mut self, tx: &Self::Item) {
+        BestTransactions::mark_invalid(&mut self.best, tx)
```

Comment:
nit: `self.best.mark_invalid(tx)` should work too?

Reply:
the impl type has the as name I can perhaps make this nicer though